test/unit/models/tasks.test.js

```diff
@@ -140,4 +145,31 @@ describe("tasks", function () {
       expect(result.taskDetails.dependsOn).to.equal(data.dependsOn);
     });
   });
+
+  describe("taskdependency", function () {
+    afterEach(function () {
+      sinon.restore();
+    });
+
+    it("should return correct result", async function () {
+      const data = {
+        taskId: "taskId1",
+        dependsOn: ["taskId2", "taskId3"],
+      };
+
+      const setStub = sinon.stub(dependencyModel.doc("taskDependencies"), "set").resolves();
+      const getStub = sinon.stub(dependencyModel.doc("taskDependencies"), "get").resolves({
+        data: () => data,
+      });
+
+      await dependencyModel.doc("taskDependencies").set(data);
+      const dependencyData = (await dependencyModel.doc("taskDependencies").get()).data();
+
+      expect(dependencyData.dependsOn).to.be.a("array");
+      expect(dependencyData.taskId).to.be.equal("taskId1");
+
+      setStub.restore();
+      getStub.restore();
+    });
+  });
```

Comment:
where is the model function being called here

Reply:
Added